src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -93,31 +93,53 @@ inline void SignalDetectorClass::doDetect()
 		}
 		else { 			
 			// Add pattern
+			uint8_t empty = 0xff;
 			if (patternLen == maxNumPattern)
 			{
 				calcHisto();
-				if (histo[patternLen] > 2) processMessage();
-				for (int16_t i = messageLen - 1; i > 0; --i)
-				{
-					if (message[i] == pattern_pos) // Finde den letzten Verweis im Array auf den Index der gleich �berschrieben wird
-					{
-						i++; // i um eins erh�hen, damit zuk�nftigen Berechnungen darauf aufbauen k�nnen
-						bufferMove(i);
+//				compress_pattern();
+
+				// search empty pattern
+				for (uint8_t p=0; p<patternLen; p++) {
```

Comment:
soweit ich das verstanden habe, sollen hier Einträge im pattern Array wiederverwendet werden.
Das neue ist, dass die zeitliche Reihenfolge, nach der die EInträge aufgenommen wurden, nicht mehr über die Position im array bestimmt werden können. Das ist in so fern interessant, da die Reihenfolge bislang eine Komponente für die Erkennung von wiederholenden Mustern war.